packages/db-base/lib/inMemFileDB.js

```diff
@@ -71,77 +71,84 @@ class InMemoryFileDB {
+        if (!fs.existsSync(parts)) {
+            fs.ensureDirSync(parts);
         }
 
-        this.datasetName = path.join(this.dataDir, this.settings.fileDB.fileName);
         this.stateTimer = null;
 
         this.backupDir = this.settings.backup.path || (path.join(this.dataDir, this.settings.fileDB.backupDirName));
 
         if (!this.settings.backup.disabled) {
-            this.zlib = this.zlib || require('zlib');
-            // Interval in minutes => to milliseconds
-            this.settings.backup.period = this.settings.backup.period === undefined ? 120 : parseInt(this.settings.backup.period);
-            if (isNaN(this.settings.backup.period)) {
-                this.settings.backup.period = 120;
-            }
-            this.settings.backup.period *= 60000;
-
-            this.settings.backup.files = this.settings.backup.files === undefined ? 24 : parseInt(this.settings.backup.files);
-            if (isNaN(this.settings.backup.files)) {
-                this.settings.backup.files = 24;
-            }
-
-            this.settings.backup.hours = this.settings.backup.hours === undefined ? 48 : parseInt(this.settings.backup.hours);
-            if (isNaN(this.settings.backup.hours)) {
-                this.settings.backup.hours = 48;
-            }
-            // Create backup directory
-            if (!fs.existsSync(this.backupDir)) {
-                fs.mkdirSync(this.backupDir);
-            }
+            this.initBackupDir();
         }
 
         this.log = tools.getLogger(this.settings.logger);
 
         this.log.debug(this.namespace + ' Data File: ' + this.datasetName);
+
         // load values from file
-        if (fs.existsSync(this.datasetName)) {
-            try {
-                this.dataset = fs.readJSONSync(this.datasetName);
-            } catch (e) {
-                this.log.error(this.namespace + ' Cannot parse ' + this.datasetName + ': ' + e.message);
-                if (fs.existsSync(this.datasetName + '.bak')) {
-                    try {
-                        this.dataset = fs.readJSONSync(this.datasetName + '.bak');
-                    } catch (e) {
-                        this.log.error(this.namespace + ' Cannot parse ' + this.datasetName + '.bak: ' + e.message);
-                        this.dataset = {};
-                    }
-                } else {
-                    this.dataset = {};
-                }
-            }
-        } else if (fs.existsSync(this.datasetName + '.bak')) {
+        this.dataset = this.loadDataset(this.datasetName);
+    }
+
+    /**
+     * Loads a dataset file
+     *
+     * @param datasetName {string} Filename of the file to load
+     * @returns {object} read data, normally as object
+     */
+    loadDatasetFile(datasetName) {
+        if (!fs.existsSync(datasetName)) {
+            throw new Error(`Database file ${datasetName} does not exists.`);
+        }
+        return fs.readJSONSync(datasetName);
```

Comment:
Can we make this async? Should be better for performance if we don't block the event loop until a multi-MB JSON was parsed.
Also least the incremental DB works completely async - or maybe that is not a problem.

Reply:
This is done exactly one time at the start of the Main js-Controller. Nothing else will happen before dBs are read and server is there. I see no issue here